packages/extension-link/src/link.ts

```diff
@@ -163,7 +163,7 @@ export const Link = Mark.create<LinkOptions>({
   renderHTML({ HTMLAttributes }) {
     // False positive; we're explicitly checking for javascript: links to ignore them
     // eslint-disable-next-line no-script-url
-    if (HTMLAttributes.href?.startsWith('javascript:')) {
+    if (HTMLAttributes.href?.toLowerCase().startsWith('javascript:')) {
```

Comment:
oof good catch

Reply:
I just tested this locally and it looks like you can fool it with leading white space too. I think that besides the protocol (e.g. `https://`) a colon is invalid so may this should be an `.includes` instead of a `.startsWith`

Reply:
It gets tricky if the URL uses basic-auth and includes the username and password (i.e., `https://javascript:my_password@example.com/` for a user named javascript would be a false positive when using `.includes`). I think it'd be better to `.trim()` before calling `.startsWith()`

Reply:
I think the correct way to do this would be to extract the scheme (everything up to the first colon), trim it, and then do a case insensitive match for `javascript`. I'll push something later today.

Reply:
Hello! Happy to see that you're looking into fixing this. ❤️ However I think there are a few issues with your approach here. 

- it would allow javascript protocol when it includes \x01-\x20 characters inside the `javascript:`
- it would allow javascript protocol when it includes whitespace characters before the colon `javascript\n:`.

It's really hard to get the blacklisting approach right. I made a PR where I try to use a whitelist approach instead, see here: https://github.com/ueberdosis/tiptap/pull/5160 It's still draft/WIP though.

I've taken inspiration from [DOMPurify](https://github.com/cure53/DOMPurify) (using it might in general be a good approach for tiptap).

I've also created test cases from PortSwiggers XSS cheat sheet: https://portswigger.net/web-security/cross-site-scripting/cheat-sheet#protocols

Reply:
Thanks for chiming in @chroth we are looking into the right approach for this sensitive topic and your contribution made us re-think some things

There are three open PRs for XSS vuln fixes:
https://github.com/ueberdosis/tiptap/pull/5157
https://github.com/ueberdosis/tiptap/pull/5160
https://github.com/ueberdosis/tiptap/pull/5153